Dockerfile

```diff
@@ -5,13 +5,12 @@ RUN mkdir -p /data/db2 \
     && chown -R mongodb:mongodb /data/db2
 
 COPY data/ /temp/
+COPY pre-processor-scripts/populate-search-fields.js /temp/
 COPY etc/ /etc/
 
 RUN mongod --fork --logpath /var/log/mongodb.log --config /etc/mongodb.conf \
     && mongoimport --jsonArray -d profiles -c gps --file /temp/gp-data-merged.json \
-    # Populate the searchWords field with words from the address and name with duplicates and punctuation removed
-    && mongo profiles --eval "db.gps.find().snapshot().forEach(function (gp) { gp.searchWords = gp.address.addressLines.concat([gp.address.postcode, gp.name]).join(' ').toLowerCase().replace(/[^\w\s]/g,'').replace(/\s+/g,' ').split(' ').filter(function(item, pos, self) { return self.indexOf(item) == pos; }).join(' '); db.gps.save(gp);} )" \
-    && mongo profiles --eval "db.gps.createIndex({ 'name': 'text', 'searchWords': 'text'}, { weights: { name: 5, searchWords: 10 } }, {name: 'SearchIndex'})" \
+    && mongo profiles temp/populate-search-fields.js \
```

Comment:
Did spend a short while seeing if we could separate the functions from the mongo specific stuff so we could test it but with no luck. The transforms should probably be done on the JSON import file outside of MongoDB but everything changes with the proposed move to Elastic Search.